resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -86,10 +86,13 @@ object ExecutorPodsSnapshot extends Logging {
             sparkContainerStatusOpt match {
               case Some(sparkContainerStatus) =>
                 sparkContainerStatus.getState.getTerminated match {
+                  // If there is no terminated state we are running.
+                  case null => PodRunning(pod)
                   case t if t.getExitCode != 0 =>
                     PodFailed(pod)
                   case t if t.getExitCode == 0 =>
                     PodSucceeded(pod)
+                  // Fall through case
                   case _ =>
```

Comment:
If we have three case statements before this line, `t is null` and `t is not null and exit code is non-zero` and `t is not null and exist code is `zero` is covered. When we fall into here?

Reply:
We shouldn't but this is to prevent the spurious "non-exhaustive match" warning.

Reply:
Oh, got it.